File: camerax_android/lib/src/recording.dart
import 'package:camerax_platform_interface/camerax_platform_interface.dart';

import 'jni.g.dart' as jni;

class MyRecording implements Recording {
  final jni.Recording jniValue;

  MyRecording(this.jniValue);

  @override
  void mute(bool muted) {
    jniValue.mute(muted);
  }

  @override
  void pause() {
    jniValue.pause();
  }

  @override
  void resume() {
    jniValue.resume();
  }

  @override
  void stop() {
    jniValue.stop();
  }
}

File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/video/Recording.kt
package dev.hebei.camerax_android.video

class Recording internal constructor(internal val obj: androidx.camera.video.Recording) {
    val isPersistent
        @androidx.camera.video.ExperimentalPersistentRecording get() = obj.isPersistent

    fun mute(muted: Boolean) {
        obj.mute(muted)
    }

    fun pause() {
        obj.pause()
    }

    fun resume() {
        obj.resume()
    }

    fun stop() {
        obj.stop()
    }

    fun close() {
        obj.close()
    }
}